File: src/solutions/java/PalindromicSubstrings.java
package solutions.java;

/*
# Problem
	- `Link`: https://leetcode.com/problems/palindromic-substrings/
# Solution
	- `Author`: Kyungtaek Lim (Jonas)
	- `Date`: Apr 20
	- `Answer`: countSubstrings
 */
public class PalindromicSubstrings {

	/*
	# Option #1
	- O(n*2)
	 */
    public static int countSubstrings(String s) {
    	int sLength = s.length();
    	int result = sLength;
    	char[] charArray = s.toCharArray();
    	int left = 0, right = 0;
    	for (int i = 1; i < sLength; i++) {
    		
    		left = i - 1;
    		right = i + 1;
    		while (left >= 0 && right < sLength && charArray[left] == charArray[right]) {
    			result++;
    			left--;
    			right++;
    		}
    		
    		left = i - 1;
    		right = i;
    		while (left >= 0 && right < sLength && charArray[left] == charArray[right]) {
    			result++;
    			left--;
    			right++;
    		}
    	}
    	return result;
    }
}

File: src/solutions/python/PalindromicSubstrings.py

'''
# Problem
	- `Link`: https://leetcode.com/problems/palindromic-substrings/
# Solution
	- `Author`: Kyungtaek Lim (Jonas)
	- `Date`: Apr 20
	- `Answer`: countSubstrings
'''
class Solution:

    '''
    # Option #1
    - O(n^2)
    '''
    def countSubStrings(self, s: str) -> int:
        len_s = len(s)
        result = len_s
        for i in range(1, len_s):
            left, right = i - 1, i + 1
            while left >= 0 and right < len_s and s[left] == s[right]:
                result += 1
                left -= 1
                right += 1
            left, right = i - 1, i
            while left >= 0 and right < len_s and s[left] == s[right]:
                result += 1
                left -= 1
                right += 1
        return result